### ISSUE TITLE
False-positive `unused-private-member` with class name

### ISSUE THREAD

**Issue by cdce8p** [PyCQA/pylint#4681]
### Steps to reproduce

```py
class Singleton:
    __instance = None

    @staticmethod
    def instance():
        if Singleton.__instance is None:
            Singleton()
        return Singleton.__instance

    def __init__(self):
        try:
            Singleton.__instance = 42  # Error here
        except Exception:  # pylint: disable=broad-except
            print("Error")
            Singleton.__instance = False  # Error here
```

**or**
```py
class Singleton:
    __instance = None

    @classmethod  # Use class method here
    def instance(cls):
        if cls.__instance is None:
            cls()
        return cls.__instance

    def __init__(self):
        try:
            Singleton.__instance = 42  # Error here
        except Exception:  # pylint: disable=broad-except
            print("Error")
            Singleton.__instance = False  # Error here
```

### Current behavior

```
test.py:12:12: W0238: Unused private member `Singleton.__instance` (unused-private-member)
test.py:15:12: W0238: Unused private member `Singleton.__instance` (unused-private-member)
```

### Expected behavior
No errors

### pylint --version output

```
pylint 2.9.4-dev0
astroid 2.6.3-dev0
Python 3.10.0b3
```

### Additional information
Probably needs to be fixed similar to #4663

**Comment by cdce8p** [comment]
@yushao2 Would you like to take a look at that?

**Comment by yushao2** [comment]
> @yushao2 Would you like to take a look at that?

Sure thing.

I don't really use classes that much, but just wanted to ask if there might be cases when a superclass's class name is used too

e.g.
```
class Singleton:
    __instance = None

    @staticmethod
    def instance():
        if Singleton.__instance is None:
            Singleton()
        return Singleton.__instance

    def __init__(self):
        try:
            Singleton.__instance = 42  # Error here
        except Exception:  # pylint: disable=broad-except
            print("Error")
            Singleton.__instance = False  # Error here

class InheritingClass(Singleton):
    def test(self):
        Singleton.__instance
```

Not sure if this is even valid syntax, but just wanted some clarifications at this stage before I start working on it

**Comment by Pierre-Sassoulas** [comment]
As ``__instance`` is private in ``Singleton`` you should not be able to access it from ``InheritingClass``

**Comment by yushao2** [comment]
> As `__instance` is private in `Singleton` you should not be able to access it from `InheritingClass`

right... not sure why my brain wasn't working. thank you :pray:

**Comment by yushao2** [comment]
@cdce8p 

picking this up after a hiatus, see #4782

**Comment by cdce8p** [comment]
> @cdce8p
> 
> picking this up after a hiatus, see #4782

I'll be sure to take a look!

**Comment by stanislavlevin** [comment]
The second example in this issue hasn't been fixed with the mentioned PR (today's main).
```
[builder@localhost BUILD]$ python3 -m pylint --version
pylint 2.10.0-dev0
astroid 2.6.5
Python 3.9.6 (default, Jun 29 2021, 10:42:27) 
[GCC 10.2.1 20210313 (ALT Sisyphus 10.2.1-alt3)]
```

**Comment by stanislavlevin** [comment]
```
[builder@localhost BUILD]$ python3 -m pylint a.py
************* Module a
a.py:12:12: W0238: Unused private member `Singleton.__instance` (unused-private-member)
a.py:15:12: W0238: Unused private member `Singleton.__instance` (unused-private-member)
```

**Comment by yushao2** [comment]
@stanislavlevin hey thanks, I'll try it out and investigate
